### ISSUE TITLE
Structural Plasticity and Spiking Networks

### ISSUE THREAD

**Issue by hdinkelbach** [ANNarchy/ANNarchy#12]
Minimal example to reproduce the error:

```
from ANNarchy import *
setup(structural_plasticity=True, seed=56789)

p0 = PoissonPopulation(10, rates=40)
p1 = Population(4, Izhikevich)

proj=Projection(p0, p1, "exc")
proj.connect_fixed_number_pre(3, weights=0.1, force_multiple_weights=True)

compile()

m = Monitor(p1, ["g_exc"])

proj.dendrite(0).prune_synapse(1)
proj.dendrite(0).prune_synapse(2)
proj.dendrite(0).prune_synapse(5)

simulate(10)

print(m.get())

```

While the forward view is correctly updated, which we could verify by comparing the connectivity matrix with/without pruning:

Matrix before the prune statements:
[[0.  0.1 0.1 0.  0.  0.1 0.  0.  0.  0. ]
 [0.1 0.  0.  0.  0.  0.  0.  0.  0.1 0.1]
 [0.  0.  0.1 0.  0.  0.1 0.  0.  0.  0.1]
 [0.  0.1 0.  0.  0.1 0.1 0.  0.  0.  0. ]]

with pruning, the first row is correctly removed:
[[0.  0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.1 0.  0.  0.  0.  0.  0.  0.  0.1 0.1]
 [0.  0.  0.1 0.  0.  0.1 0.  0.  0.  0.1]
 [0.  0.1 0.  0.  0.1 0.1 0.  0.  0.  0. ]]

However, when we analyze the stored conductance, we notice that there is still a weight transmission. So I assume, that the inverse view is not updated correctly.

**Comment by Thanh-Binh** [comment]
Hello
I am looking at the code for pruning() and found a possible bug here:
### Original ###
           for(int i = 0; i < post_rank.size(); i++){
                int rk_post = post_rank[i];
                for(int j = 0; j < pre_rank[i].size(); j++){
                    int rk_pre = pre_rank[i][j];
                    if((%(condition)s)%(proba)s){
                        removeSynapse(i, j);
                    }
                }
            }

It should be changed to:

           for(int i = 0; i < post_rank.size(); i++){
                int rk_post = post_rank[i];
                for(int j = 0; j < pre_rank[i].size(); ){
                    int rk_pre = pre_rank[i][j];
                    if((%(condition)s)%(proba)s){
                        removeSynapse(i, j);
                    }
                    else ++j;
                }
            }

What do you think?

**Comment by hdinkelbach** [comment]
Hi,

Thanks for your bug report. Indeed, the index of j is updated not correctly if multiple synapses are removed. Currently, the removal of a synapse would skip the next one ...

I need to think shortly about your proposed solution, on the first view it looks good.

**Comment by hdinkelbach** [comment]
Fixed in 4.8.0.
